src/pages/widget/Subscribe/index.tsx

```diff
@@ -52,17 +53,29 @@ const WidgetSubscribe: React.FC = () => {
   }, [pushClientProxy, dappOrigin, dappIcon, dappName, dappNotificationDescription, userPubkey])
 
   useEffect(() => {
-    const dappSub = activeSubscriptions.find(sub => sub.metadata.url === dappOrigin)
-    if (dappSub) {
-      setTimeout(() => {
-        const communicator = new JsCommunicator(new EventEmitter())
+    const navIfSubFound = () => {
+      const dappSub = activeSubscriptions.find(sub => sub.metadata.url === dappOrigin)
+      if (dappSub) {
+        const communicator = new JsCommunicator(emitter)
         communicator.postToExternalProvider('dapp_subscription_settled', {}, 'notify')
-      }, 10)
-      setTimeout(() => {
         nav(`/notifications/${dappSub.topic}`)
-      }, 0)
+      }
     }
-  }, [activeSubscriptions, nav])
+
+    navIfSubFound()
+
+    /*
+     * The following two event listeners are essentially fail-safes if something goes wrong
```

Comment:
What is "something goes wrong" in this case and do we know why? I'd prefer to figure out root cause since sticking on a band aid isn't the way forward long term.